src/averageAges.js

```diff
@@ -54,9 +70,18 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
-}
+const calculateAverageAgeDiff = (people, onlyWithSon) => {
+  const childrenArray = people.filter(({ mother, sex }) => onlyWithSon
+    ? sex === 'm' && mother && people.some(({ name }) => name === mother)
+    : mother && people.some(({ name }) => name === mother));
```

Comment:
This code hard to read, better separate this code to 2 variables. 

Do not forget [checklist points 2, 3](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)